src/tree/updater_gpu_hist.cu

```diff
@@ -1182,42 +1182,35 @@ class GPUHistMakerSpecialised{
   }
 
   void ApplySplit(const ExpandEntry& candidate, RegTree* p_tree) {
-    // Add new leaves
     RegTree& tree = *p_tree;
-    tree.ExpandNode(candidate.nid, candidate.split.findex, candidate.split.fvalue,
-                    candidate.split.dir == kLeftDir);
-    auto& parent = tree[candidate.nid];
-    tree.Stat(candidate.nid).loss_chg = candidate.split.loss_chg;
-
-    // Set up child constraints
-    node_value_constraints_.resize(tree.GetNodes().size());
     GradStats left_stats(param_);
     left_stats.Add(candidate.split.left_sum);
     GradStats right_stats(param_);
     right_stats.Add(candidate.split.right_sum);
-    node_value_constraints_[candidate.nid].SetChild(
-        param_, parent.SplitIndex(), left_stats, right_stats,
-        &node_value_constraints_[parent.LeftChild()],
-        &node_value_constraints_[parent.RightChild()]);
-
-    // Configure left child
+    GradStats parent_sum(param_);
+    parent_sum.Add(left_stats);
+    parent_sum.Add(right_stats);
+    node_value_constraints_.resize(tree.GetNodes().size());
+    auto base_weight = node_value_constraints_[candidate.nid].CalcWeight(param_, parent_sum);
```

Comment:
Do we need to re-compute `base_weight` here? Is it possible to re-use the leaf weight of `nid` node prior to expanding?

Reply:
The leaf weight is multiplied by learning rate, we would have to undo this with potential numerical issues.

Reply:
Ah I see. Then let's proceed as it is. Go ahead and merge?